csi.proto

```diff
@@ -510,21 +562,29 @@ message NodePublishVolumeRequest {
   // this capability. This is an OPTIONAL field.
   map<string, string> publish_volume_info = 3;
 
+  // The path to which the device was mounted by `NodePublishDevice`.
+  // It MUST be an absolute path in the root filesystem of the process
+  // serving this request.
+  // It MUST be set if the Node Plugin implements the 
+  // `PUBLISH_UNPUBLISH_DEVICE` node capability.
+  // This is an OPTIONAL field.
+  string global_target_path = 4;
+
   // The path to which the volume will be published. It MUST be an
   // absolute path in the root filesystem of the process serving this
   // request. The CO SHALL ensure uniqueness of target_path per volume.
   // The CO SHALL ensure that the path exists, and that the process
   // serving the request has `read` and `write` permissions to the path.
   // This is a REQUIRED field.
-  string target_path = 4;
+  string target_path = 5;
```

Comment:
should we avoid renumbering proto fields now that we have a tagged/published version of the spec in the world?

Reply:
My understanding that this should be an OK change, it will introduce incompatablity between different CSI spec versions but I think thats OK. Thats why we have a supported_versions field right?

The main issue with changing proto tags is when the protos are actually persisted somewhere and new code with different tags comes along and reads the old protos and gets confused. I don't believe that is an issue here.

@saad-ali may be able to provide more context.

Reply:
Client/server mismatch of proto field number can cause issues. However, for v0.2, we should be ok making breaking changes, if needed.